app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -343,21 +345,29 @@ private static PokemonNameCorrector initCorrectorFromPrefs(PokeInfoCalculator po
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        if (intent != null && intent.hasExtra(KEY_TRAINER_LEVEL)) {
-            trainerLevel = intent.getIntExtra(KEY_TRAINER_LEVEL, 1);
-            statusBarHeight = intent.getIntExtra(KEY_STATUS_BAR_HEIGHT, 0);
-            batterySaver = intent.getBooleanExtra(KEY_BATTERY_SAVER, false);
-            makeNotification(this);
-            createInfoLayout();
-            createIVButton();
-            createArcPointer();
-            createArcAdjuster();
-            /* Assumes MainActivity initialized ScreenGrabber before starting this service. */
-            if (!batterySaver) {
-                screen = ScreenGrabber.getInstance();
-                startPeriodicScreenScan();
-            } else {
-                screenShotHelper = ScreenShotHelper.start(Pokefly.this);
+        if (intent != null) {
+            if (ACTION_STOP.equals(intent.getAction())) {
+                if (screen != null) {
+                    screen.exit();
```

Comment:
Couple of options:

1. Move the exit from MainActivity to here, then both intents would trigger same path? Not ideal, but better than 2 exits() I think.

2. Cleanest, but not easiest to figure out: Figure out a good way to move screen.init() to Pokefly. I was too confused with the existing code that does a lot of intent passing and request codes to get the projection manager  (or whatever it was called) to start recording. I don't think it mandates the need of an Activity to be able to get that to work. It might just need Context (which both Service and Activity are). Also, the code was in a much worse state when I created ScreenGrabber -- so I didn't move it screen init to Pokefly.